src/main/java/com/cloudbees/jenkins/plugins/bitbucket/hooks/WebhookAutoRegisterListener.java

```diff
@@ -199,7 +201,7 @@ private void removeHooks(SCMSourceOwner owner) throws IOException, InterruptedEx
                 BitbucketWebHook hook = null;
                 for (BitbucketWebHook h : existent) {
                     // Check if there is a hook pointing to us
-                    if (h.getUrl().equals(Jenkins.getActiveInstance().getRootUrl() + BitbucketSCMSourcePushHookReceiver.FULL_PATH)) {
+                    if (h.getUrl().startsWith(Jenkins.getActiveInstance().getRootUrl() + BitbucketSCMSourcePushHookReceiver.FULL_PATH)) {
```

Comment:
What's the reason behind this change?

Reply:
The native webhooks don't include the Bitbucket server URL anywhere in the payload. The only way I came up with to get that piece of information was to introduce a query parameter to the webhook's callback URL. So I simply changed it here to be a prefix match, so that URLs with query parameters also satisfy the condition. One could add a more thorough check here, but I felt this simple one was sufficient.